cmd/juju/application/deployer/deployer.go

```diff
@@ -425,6 +429,40 @@ func (d *factory) charmStoreCharm() (Deployer, error) {
 	}, nil
 }
 
+func resolveCharmURL(path string) (*charm.URL, error) {
+	// If the charmhub integration is in effect, then we ensure that schema is
+	// defined and we can parse the URL correctly.
+	if featureflag.Enabled(feature.CharmHubIntegration) {
```

Comment:
We'll have to take care here when removing the feature flag.  A TODO to remind us this needs to be rejiggered, not just remove the feature flag?